src/placement/pl_map_common.c

```diff
@@ -217,65 +215,8 @@ remap_list_fill(struct pl_map *map, struct daos_obj_md *md,
 			 * for rebuild, perhaps they should be unified.
 			 */
 			if (l_shard->po_shard != -1) {
-				struct pool_target      *target;
-				int                      leader;
-
 				D_ASSERT(f_shard->fs_tgt_id != -1);
 				D_ASSERT(*idx < array_size);
-
-				/* If the caller does not care about DTX related
-				 * things (myrank == -1), then fill it directly.
-				 */
-				if (myrank == -1)
-					goto fill;
-
-				leader = pl_select_leader(md->omd_id,
-					l_shard->po_shard, layout->ol_nr,
-					true, pl_obj_get_shard, layout);
-
-				if (leader < 0) {
-					D_WARN("Not sure whether current shard "
-					       "is leader or not for obj "
-					       DF_OID", fseq:%d, status:%d, "
-					       "ver:%d, shard:%d, rc = %d\n",
-					       DP_OID(md->omd_id),
-					       f_shard->fs_fseq,
-					       f_shard->fs_status, r_ver,
-					       l_shard->po_shard, leader);
-					goto fill;
-				}
-
-				rc = pool_map_find_target(map->pl_poolmap,
-							  leader, &target);
-				D_ASSERT(rc == 1);
-
-				if (myrank != target->ta_comp.co_rank) {
-					/* The leader shard is not on current
-					* server, then current server cannot
-					 * know whether DTXs for current shard
-					 * have been re-synced or not. So skip
-					 * the shard that will be handled by
-					 * the leader on another server.
-					 */
-					D_DEBUG(DB_PL, "Current replica (%d)"
-						"isn't the leader (%d) for obj "
-						DF_OID", fseq:%d, status:%d, "
-						"ver:%d, shard:%d, skip it\n",
-						myrank, target->ta_comp.co_rank,
-						DP_OID(md->omd_id),
-						f_shard->fs_fseq,
-						f_shard->fs_status,
-						r_ver, l_shard->po_shard);
-					continue;
-				}
-
-fill:
-				D_DEBUG(DB_PL, "Current replica (%d) is the "
-					"leader for obj "DF_OID", fseq:%d, "
-					"ver:%d, shard:%d, to be rebuilt.\n",
-					myrank, DP_OID(md->omd_id),
-					f_shard->fs_fseq,
-					r_ver, l_shard->po_shard);
```

Comment:
why remove above section of code? (just to confirm, thx)

Reply:
oh, these are for eliminate the non-leader scan, which is wrong for ec object. Just discussed with Nasf, we should do it in a different way even for replicate object.